services/m3dbnode/server/serve.go

```diff
@@ -51,57 +51,72 @@ func DefaultShardSet() (sharding.ShardSet, error) {
 	return sharding.NewShardSet(shards, sharding.DefaultHashGen(1024))
 }
 
-// DefaultNamespaces creates a list of default namespaces.
-func DefaultNamespaces() []namespace.Metadata {
-	opts := namespace.NewOptions()
-	return []namespace.Metadata{
-		namespace.NewMetadata(ts.StringID(defaultNamespaceName), opts),
+// DefaultTopologyInitializer creates a default topology initializer
+func DefaultTopologyInitializer(
+	hostID string,
+	tchannelNodeAddr string,
+) (topology.Initializer, error) {
+	shardSet, err := DefaultShardSet()
+	if err != nil {
+		return nil, err
 	}
+
+	return DefaultTopologyInitializerForShardSet(hostID, tchannelNodeAddr, shardSet)
 }
 
-// DefaultClientOptions creates a default m3db client options.
-func DefaultClientOptions(id, tchannelNodeAddr string, shardSet sharding.ShardSet) (client.Options, error) {
+// DefaultTopologyInitializerForShardSet creates a default topology initializer for a shard set
+func DefaultTopologyInitializerForShardSet(
+	hostID string,
+	tchannelNodeAddr string,
+	shardSet sharding.ShardSet,
+) (topology.Initializer, error) {
 	var localNodeAddr string
 	if !strings.ContainsRune(tchannelNodeAddr, ':') {
 		return nil, errors.New("tchannelthrift address does not specify port")
 	}
 	localNodeAddrComponents := strings.Split(tchannelNodeAddr, ":")
 	localNodeAddr = fmt.Sprintf("127.0.0.1:%s", localNodeAddrComponents[len(localNodeAddrComponents)-1])
 
-	hostShardSet := topology.NewHostShardSet(topology.NewHost(id, localNodeAddr), shardSet)
+	hostShardSet := topology.NewHostShardSet(topology.NewHost(hostID, localNodeAddr), shardSet)
 	staticOptions := topology.NewStaticOptions().
 		SetShardSet(shardSet).
 		SetReplicas(1).
 		SetHostShardSets([]topology.HostShardSet{hostShardSet})
 
-	return client.NewOptions().SetTopologyInitializer(topology.NewStaticInitializer(staticOptions)), nil
+	return topology.NewStaticInitializer(staticOptions), nil
```

Comment:
Should the topology initializer be a dynamic one for dynamic resharding? it should take in the Config Service client like the M3DB client, or even have them both take in the dynamic topology directly?